src/sockaddr.rs

```diff
@@ -12,24 +12,68 @@ use winapi::shared::ws2ipdef::SOCKADDR_IN6_LH_u;
 /// The address of a socket.
 ///
 /// `SockAddr`s may be constructed directly to and from the standard library
-/// `SocketAddr`, `SocketAddrV4`, and `SocketAddrV6` types.
+/// [`SocketAddr`], [`SocketAddrV4`], and [`SocketAddrV6`] types.
 pub struct SockAddr {
     storage: sockaddr_storage,
     len: socklen_t,
 }
 
+#[allow(clippy::len_without_is_empty)]
 impl SockAddr {
+    /// Create a `SockAddr` from the underlying storage and its length.
+    ///
+    /// The type of the address storage and length passed to the function `init`
+    /// is OS/architecture specific.
+    ///
+    /// # Safety
+    ///
+    /// - The `SockAddr`'s address family must match its contents.
+    /// - The given length of the `SockAddr` must be correct.
+    ///
+    /// # Examples
+    ///
+    #[cfg_attr(target_os = "linux", doc = "```")]
+    #[cfg_attr(not(target_os = "linux"), doc = "```ignore")]
```

Comment:
Can you change the example to something that works on all OSs? You can use the same one from `Socket::init`.